src/panels/developer-tools/service/developer-tools-service.ts

```diff
@@ -275,6 +280,14 @@ class HaPanelDevService extends LitElement {
     try {
       await callExecuteScript(this.hass, [this._serviceData]);
     } catch (err) {
+      const [domain, service] = this._serviceData.service.split(".", 2);
+      if (
+        err.error.code === ERR_CONNECTION_LOST &&
+        serviceCallWillDisconnect(domain, service)
```

Comment:
This can also work:

```suggestion
      if (
        err.error.code === ERR_CONNECTION_LOST &&
        serviceCallWillDisconnect(...this._serviceData.service.split(".", 2))
```

(but is less readable)